features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2

```diff
@@ -829,16 +829,34 @@
             <!--
                 FAPI Profile Support.
             -->
-            {% if oauth.oidc.fapi is defined %}
             <FAPI>
                 {% if oauth.oidc.fapi.enable_ciba_profile is defined %}
                 <EnableCibaProfile>{{oauth.oidc.fapi.enable_ciba_profile}}</EnableCibaProfile>
                 {% endif %}
                 {% if oauth.oidc.fapi.enable_security_profile is defined %}
                 <EnableSecurityProfile>{{oauth.oidc.fapi.enable_security_profile}}</EnableSecurityProfile>
                 {% endif %}
+                <SupportedClientAuthenticationMethods>
+                    {% if oauth.oidc.fapi.client_authentication_methods|length %}
+                        {% for authmethod in oauth.oidc.fapi.client_authentication_methods %}
+                            <SupportedClientAuthenticationMethod>{{authmethod}}</SupportedClientAuthenticationMethod>
+                        {% endfor %}
+                    {% else %}
```

Comment:
Instead of having a else if we can add the defaults to the default.json so this conditional check is auto-handled. If no configs are defined the default.json config will apply here.

Reply:
Fixed in 572230c036c602e133e13c0aeb2ab88f7e3a74f1